interactions/api/cache.py

```diff
@@ -38,8 +40,13 @@ class Storage(Generic[_T]):
     def __repr__(self) -> str:
         return f"<{self.__class__.__name__} object containing {len(self.values)} items.>"
 
-    def __init__(self) -> None:
-        self.values: Dict["Key", _T] = {}
+    def __init__(self, limit: Optional[int] = float("inf")) -> None:
+        """
+
+        :param limit: The maximum number of items to store
+        :type limit: Optional[int]
```

Comment:
```suggestion
        :param Optional[int] limit: The maximum number of items to store
```
To account for the new way to write docstrings